fineract-provider/src/main/java/org/apache/fineract/accounting/journalentry/service/AccrualBasedAccountingProcessorForSavings.java

```diff
@@ -182,9 +182,21 @@ else if (savingsTransactionDTO.getTransactionType().isInterestPosting()) {
             else if (savingsTransactionDTO.getTransactionType().isAccrual()) {
                 // Post journal entry for Accrual Recognition
                 if (savingsTransactionDTO.getAmount().compareTo(BigDecimal.ZERO) > 0) {
-                    this.helper.createCashBasedJournalEntriesAndReversalsForSavings(office, currencyCode,
-                            AccrualAccountsForSavings.INTEREST_ON_SAVINGS.getValue(), AccrualAccountsForSavings.INTEREST_PAYABLE.getValue(),
-                            savingsProductId, paymentTypeId, savingsId, transactionId, transactionDate, amount, isReversal);
+                    if (feePayments.size() > 0 || penaltyPayments.size() > 0) {
+                        this.helper.createCashBasedJournalEntriesAndReversalsForSavings(office, currencyCode,
```

Comment:
Free and penalty is not the same, we should not use the same accounting for both!